tests/tasks/sequencing_publisher/test_main.py

```diff
@@ -0,0 +1,373 @@
+        )
+
+    if role_types_entries:
+        write_to_table(
+            database=mlwh_events_database,
+            insert_into_table_query="""
+                INSERT INTO role_types (id, `key`) VALUES (%s, %s);
+            """,
+            entries=role_types_entries,
+        )
+
+    if subjects_entries:
+        write_to_table(
+            database=mlwh_events_database,
+            insert_into_table_query="""
+                INSERT INTO subjects (id, uuid, friendly_name) VALUES (%s, %s, %s);
+            """,
+            entries=subjects_entries,
+        )
+
+
+@patch("janitor.rabbitmq.rabbit.Rabbit.batch_publish_messages", return_value=None)
+@patch("janitor.helpers.mlwh_helpers.make_sample_sequence_message_dicts")
+def test_given_test_data_in_databases_when_querying_database_then_check_correct_rows_returned(
+    mock_publish, mock_make_messages, mock_info, mock_error, config, mlwh_database, mlwh_events_database, mock_rabbit
+):
+    write_to_mlwh_database(
+        mlwh_database=mlwh_database,
+        iseq_flowcell_entries=test_entries.good_input_entry["iseq_flowcell"],
+        iseq_product_metrics_entries=test_entries.good_input_entry["iseq_product_metrics"],
+        iseq_run_status_entries=test_entries.good_input_entry["iseq_run_status"],
+        sample_entries=test_entries.good_input_entry["sample"],
+        seq_product_irods_locations_entries=test_entries.good_input_entry["seq_product_irods_locations"],
+        study_entries=test_entries.good_input_entry["study"],
+    )
+
+    write_to_mlwh_events_database(
+        mlwh_events_database=mlwh_events_database,
+        events_entries=test_entries.good_input_entry["events"],
+        event_types_entries=test_entries.good_input_entry["event_types"],
+        roles_entries=test_entries.good_input_entry["roles"],
+        role_types_entries=test_entries.good_input_entry["role_types"],
+        subjects_entries=test_entries.good_input_entry["subjects"],
+    )
+
+    expected_message_dicts = [
+        SampleSequenceMessage(
+            change_date=datetime(2023, 9, 11, 9, 10, 43),
+            id_run=47819,
+            sequencing_study="sequencing_study",
+            sample_supplier_id="sample_supplier_id",
+            labware_barcode="labware_barcode",
+            run_status=1,
+            irods_root_collection="root_collection",
+            irods_data_relative_path="data_relative",
+            irods_secondary_data_relative_path=None,
+            latest_timestamp=datetime(2023, 9, 11, 9, 10, 43),
+        )
+    ]
+
+    with patch("janitor.helpers.log_helpers.load_job_timestamp") as mock_latest_timestamp:
+        mock_latest_timestamp.return_value = None
+        get_and_publish_sequencing_run_status_changes(config)
+        assert mock_make_messages.has_calls(call(expected_message_dicts))
+
+    assert mock_info.has_calls(
+        call("Starting sequencing publisher task..."),
+        call("Closing connections to databases..."),
```

Comment:
If we change the logs we may break this test. Is this check really needed?

Reply:
Probably not, I think I'll remove it because it's not the main focus of the test and the tests should check the task behaves in the expected way anyways